components/script/dom/gpudevice.rs

```diff
@@ -636,6 +650,86 @@ impl GPUDeviceMethods for GPUDevice {
         GPUCommandEncoder::new(&self.global(), self.channel.clone(), encoder, true)
     }
 
+    /// https://gpuweb.github.io/gpuweb/#dom-gpudevice-createtexture
+    fn CreateTexture(&self, descriptor: &GPUTextureDescriptor) -> DomRoot<GPUTexture> {
+        let mut valid = true;
+        let size = match descriptor.size {
+            GPUExtent3D::GPUExtent3DDict(ref dict) => wgt::Extent3d {
+                width: dict.width,
+                height: dict.height,
+                depth: dict.depth,
+            },
+            GPUExtent3D::RangeEnforcedUnsignedLongSequence(ref v) => wgt::Extent3d {
+                width: v[0],
+                height: v[1],
+                depth: v[2],
+            },
+        };
+        let desc = wgt::TextureDescriptor {
+            label: Default::default(),
+            size,
+            mip_level_count: descriptor.mipLevelCount,
+            sample_count: descriptor.sampleCount,
+            dimension: match descriptor.dimension {
+                GPUTextureDimension::_1d => wgt::TextureDimension::D1,
+                GPUTextureDimension::_2d => wgt::TextureDimension::D2,
+                GPUTextureDimension::_3d => wgt::TextureDimension::D3,
+            },
+            format: convert_texture_format(descriptor.format),
+            usage: match wgt::TextureUsage::from_bits(descriptor.usage) {
+                Some(t) => t,
+                None => {
+                    valid = false;
+                    wgt::TextureUsage::empty()
+                },
+            },
+        };
+
+        let texture_id = self
+            .global()
+            .wgpu_id_hub()
+            .lock()
+            .create_texture_id(self.device.0.backend());
+
+        self.channel
+            .0
+            .send(WebGPURequest::CreateTexture {
+                device_id: self.device.0,
+                texture_id,
+                descriptor: desc,
+            })
+            .expect("Failed to create WebGPU Texture");
+
+        let texture = webgpu::WebGPUTexture(texture_id);
+
+        let texture_size = match descriptor.size {
```

Comment:
you already have this match above :)

Reply:
The difference is the type that is returned by the arms. I'll try to store a `GPUTextureDescriptor` instead, as you mentioned. We won't need this in that case.

Reply:
I guess I would still need this separately. Or is there a better way to do this?

Reply:
It looks like you could have it in two stages: `GPUExtent3D -> GPUExtent3DDict`, and separately `GPUExtent3DDict -> wgpu::Extent3d`

Reply:
Okay, so we'll have a separate method for each. And for `GPUExtent3D -> wgt::Extent3d` we'll have to call both of them, right?

Reply:
correct, yes

Reply:
I am usually bad with names :P, but here's what I am thinking of naming the methods- `convert_texture_size_to_dict` for the first one and `convert_texture_size_to_wgt` for the second. I'll change them to something else if you say so. Will save me the effort to push the changes again :)

Reply:
that sounds ok to me :)